yt_dlp/extractor/triller.py

```diff
@@ -0,0 +1,309 @@
+        video_uuid = video_info.get('video_uuid')
+        video_id = video_info.get('id')
+
+        formats = []
+        video_url = traverse_obj(video_info, 'video_url', 'stream_url')
+        if video_url:
+            formats.append({
+                'url': video_url,
+                'ext': 'mp4',
+                'vcodec': 'h264',
+                'width': video_info.get('width'),
+                'height': video_info.get('height'),
+                'format_id': url_basename(video_url).split('.')[0],
+                'filesize': video_info.get('filesize'),
+            })
+        video_set = video_info.get('video_set', [])
+        for video in video_set:
+            resolution = video.get('resolution') or ''
+            formats.append({
+                'url': video['url'],
+                'ext': 'mp4',
+                'vcodec': video.get('codec'),
+                'vbr': int_or_none(video.get('bitrate'), 1000),
+                'width': int_or_none(resolution.split('x')[0]),
+                'height': int_or_none(resolution.split('x')[1]),
+                'format_id': url_basename(video['url']).split('.')[0],
+            })
+        audio_url = video_info.get('audio_url')
+        if audio_url:
+            formats.append({
+                'url': audio_url,
+                'ext': 'm4a',
+                'format_id': url_basename(audio_url).split('.')[0],
+            })
+
+        manifest_url = video_info.get('transcoded_url')
+        if manifest_url:
+            formats.extend(self._extract_m3u8_formats(
+                manifest_url, video_id, 'mp4', entry_protocol='m3u8_native',
+                m3u8_id='hls', fatal=False, note=False))
+        self._sort_formats(formats)
+
+        comment_count = int_or_none(video_info.get('comment_count'))
+
+        if not user_info:
+            user_info = traverse_obj(video_info, 'user', default={})
+
+        username = user_info.get('username')
+
+        return {
+            'id': str_or_none(video_id) or video_uuid,
+            'title': video_info.get('description') or f'Video by {username}',
+            'thumbnail': video_info.get('thumbnail_url'),
+            'description': video_info.get('description'),
+            'uploader': str_or_none(username),
+            'uploader_id': str_or_none(user_info.get('user_id')),
+            'creator': str_or_none(user_info.get('name')),
+            'timestamp': unified_timestamp(video_info.get('timestamp')),
+            'upload_date': unified_strdate(video_info.get('timestamp')),
+            'duration': int_or_none(video_info.get('duration')),
+            'view_count': int_or_none(video_info.get('play_count')),
+            'like_count': int_or_none(video_info.get('likes_count')),
+            'artist': str_or_none(video_info.get('song_artist')),
+            'track': str_or_none(video_info.get('song_title')),
+            'webpage_url': f'https://triller.co/@{username}/video/{video_uuid}',
```

Comment:
This makes `username` a mandatory field. Is that intentional?

Reply:
`username` is always capturable via the url

Reply:
but you aren't doing that. See L124

Reply:
good catch, thanks. fixed